rest-api/dao/participant_summary_dao.py

```diff
@@ -336,6 +394,67 @@ def calculate_enrollment_status(self, consent_for_study_enrollment_and_ehr,
       return EnrollmentStatus.MEMBER
     return EnrollmentStatus.INTERESTED
 
+  def calculate_member_time(self, consent_for_study_enrollment_and_ehr, participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.enrollmentStatusMemberTime is not None:
+      return participant_summary.enrollmentStatusMemberTime
+    elif consent_for_study_enrollment_and_ehr:
+      return participant_summary.consentForElectronicHealthRecordsTime
+    else:
+      return None
+
+  def calculate_core_stored_sample_time(self, consent_for_study_enrollment_and_ehr,
+                                        participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.numCompletedBaselinePPIModules == \
+      self._get_num_baseline_ppi_modules() and \
```

Comment:
Check the relevant inputs first, only calculate sample times when all other criteria meet.